src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotMatchKeywords.cs

```diff
@@ -95,28 +96,28 @@ public sealed class IdentifiersShouldNotMatchKeywordsAnalyzer : DiagnosticAnalyz
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            // TODO: Make this analyzer thread-safe.
-            //analysisContext.EnableConcurrentExecution();
-
+            analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            var keywordNamedNamespaces = new HashSet<string>();
+            analysisContext.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var keywordNamedNamespaces = new ConcurrentDictionary<string, string>();
```

Comment:
Thanks, I've added that in https://github.com/dotnet/roslyn-analyzers/pull/991/commits/f104454e3974ba8cb66fbcbf6179b7d7dad26635.